draft-ietf-taps-interface.md

```diff
@@ -705,6 +706,65 @@ ChallengeCallback := NewCallback({
 SecurityParameters.SetIdentityChallengeCallback(challengeCallback)
 ~~~
 
+## Specifying Caching Context {#cache-context}
+
+During the lifetime of Connection, the various protocols used may generate 
+state or statistics that can be cached to help optimize future Connections.
+New Connections can then take advantage of this cached state to influence
+connection establishment, tuning protocol heuristics, and resuming state.
+
+Some examples of cached state include:
+
+- DNS query answers may be cached for use in subsequent Connections
+to the same hostname.
+
+- TCP round-trip-time (RTT) metrics may be cached to tune delays for
+protocol stack racing.
+
+- TLS state may be cached for use in session resumption.
+
+Within a single application, there may need to be several distinct and separate
+caches to ensure that data and state is not shared between certain Connections.
+Each Preconnection is therefore associated with a CacheContext, which defines
+a distinct set of cached information. All Connections and Preconnections that are
```

Comment:
Possibly add here, or in the implementation draft: Moreover, certain cache state may not be appropriate for re-use for all applications. For example, to improve privacy and limit client tracking, TLS 1.3 {{!RFC8446}} connection state should not be re-used absent early data to write, as this introduces an unnecessary tracking vector without saving any round trips. (It may however cost more CPU.)

Reply:
In Bangkok we said that the default is that all sharing is allowed: 
If the application does not provide any explicit caching context, all Connections are associated with the same implicit caching context, so they can share state.

If we do it like this, let's make it very explicit in the text:
I suggest to move the sentence "By default, a Preconnection is associated with an implicit CacheContext." up here, clarify that it's always the same implicit CacheContext, and say that this means that by default, all Connections within the same application context are allowed to share state. 
After this, the warning about sharing TLS 1.3 connection state fits quite nicely imo.

Alternatively, we could set the default that all state can be shared _except_ TLS 1.3 connection state ...?